src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -173,62 +172,34 @@ private Future<Void> getPiecesByIds(List<String> ids, Map<String, List<Piece>> p
    * Validates if the piece corresponds to PO Line specified in the request and
    * can be received/checked-in. If all checks pass, adds to map
    *
-   * @param piece
-   *          {@link Piece} piece to validate and add to map
-   * @param piecesByPoLine
-   *          map with PO line id as a key and list of corresponding pieces as a
-   *          value
+   * @param piece          {@link Piece} piece to validate and add to map
+   * @param piecesByPoLine map with PO line id as a key and list of corresponding pieces as a
+   *                       value
    */
   private void addPieceIfValid(Piece piece, Map<String, List<Piece>> piecesByPoLine) {
     String poLineId = piece.getPoLineId();
     String pieceId = piece.getId();
 
     // Validate if the piece actually corresponds to PO line specified in the
     // request
-    if (piecesByLineId.containsKey(poLineId) && piecesByLineId.get(poLineId).containsKey(pieceId)) {
-      // Validate if the piece is not yet received
-      if (piece.getReceivingStatus() != ReceivingStatus.RECEIVED || isRevertToOnOrder(piece)) {
-        piecesByPoLine.computeIfAbsent(poLineId, v -> new ArrayList<>())
-          .add(piece);
-      } else {
-        addError(poLineId, pieceId, PIECE_ALREADY_RECEIVED.toError());
-      }
-    } else {
+    if (!piecesByLineId.containsKey(poLineId) || !piecesByLineId.get(poLineId).containsKey(pieceId)) {
       addError(getPoLineIdByPieceId(pieceId), pieceId, PIECE_POL_MISMATCH.toError());
+      return;
```

Comment:
Currently, it is skipping other checks if it found error. Maybe it is better to check all validation at once based on old implementation (if am not mistake)

Reply:
This changed is from first commit - nothing changed except that method was moved to make proper grouping